gnd/src/main/java/com/google/android/gnd/ui/map/MapAdapter.java

```diff
@@ -30,15 +32,18 @@
 public interface MapAdapter {
 
   /** Returns a stream that emits map pins clicked by the user. */
+  @Hot
```

Comment:
Good question.. according to this, we're probably misusing them:
https://github.com/ReactiveX/RxJava/wiki/What's-different-in-2.0#which-type-to-use has this to say

That said, I'm a bit confused, because this guidance doesn't mention that back pressure LATEST is still a reasonable strategy for large stateless sequences, even when you can't throttle the upstream producer.

In fact, I can't think of any reason not to _always_ use Flowable with back pressure LATEST for stateless sequences.

`LiveDataReactiveStreams.fromPublisher()` only accepts `Publisher` (`Flowable`), and RxFirestore and Room impls also return them. I'll try to apply Flowable here as well since it seems to make more sense. Please cmiiw.

Reply:
Actually, for UI events, debouncing seems to be the recommended strategy. The docs linked above say we should use Observable if:

>You deal with GUI events such as mouse moves or touch events: these can rarely be backpressured reasonably and aren't that frequent. You may be able to handle an element frequency of 1000 Hz or less with Observable but consider using sampling/debouncing anyway.

Reply:
Three's a charm: In the cases where I applied `Flowable` there can be many events returned in the stream. Since we're interested in consuming as many of them as we can (we want the map to pan smoothly), debounce isn't appropriate here. Instead, it appears `Flowable` would be the right thing here. As always, please cmiiw!

Reply:
I think `Flowable` might be more appropriate here as you are saying but I have to read a bit more, this doc seems to have a good explanation: https://github.com/ReactiveX/RxJava/wiki/Backpressure-(2.0)

Reply:
That's a great resource, adding it to our WIP doc, thanks! Yeah, in this case it sounds like `Flowable` is appropriate since there's a chance the UI can't keep up, whereas in other cases it's unlikely for that to happen.

Relevant discussion on why we wouldn't just always use `Flowable`:
https://stackoverflow.com/questions/41609463/why-not-always-use-flowable-in-rxjava2